src/components/TextInputAutoGrow.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import {View, Text} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {getAutoGrowTextInputStyle, getHiddenElementOutsideOfWindow} from '../styles/styles';
+import withWindowDimensions, {windowDimensionsPropTypes} from './withWindowDimensions';
+import TextInputFocusable from './TextInputFocusable';
+
+const propTypes = {
+
+    // The value of the comment box
+    value: PropTypes.string.isRequired,
+
+    // A ref to forward to the text input
+    forwardedRef: PropTypes.func.isRequired,
+
+    // General styles to apply to the text input
+    // eslint-disable-next-line react/forbid-prop-types
+    inputStyle: PropTypes.object,
+
+    // Styles to apply to the text input
+    // eslint-disable-next-line react/forbid-prop-types
+    textStyle: PropTypes.object.isRequired,
+
+    /* Window Dimensions Props */
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    inputStyle: {},
+};
+
+class TextInputAutoGrow extends React.Component {
```

Comment:
I feel like this component has a confusing name since the `TextInputFocusable` component does "auto grow" by setting `numLines`. This component looks like more of a "variable width text input", but "auto grow" is ambigous.

Not sure if others feel the same way. I wasn't too sure why we need this component or what it does until digging into the logic.

Reply:
I renamed it as `TextInputAutoWidth`. I think the component's behaviour is more understandable now.